TMDB Movies/Controllers/List/TMDBListViewController.swift

```diff
@@ -0,0 +1,131 @@
+    //MARK: Lifecycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = "Upcoming"
+        let movies = staticMovies
+        guard movies.isEmpty else {
+            datasource.items = movies
+            reloadData()
+            return
+        }
+        makeRequest()
+    }
+    
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        DispatchQueue.main.async { [weak self] in
+            self?.collectionView.collectionViewLayout.invalidateLayout()
+            self?.collectionView.reloadData()
+        }
+    }
+    
+    //MARK: Setup
+    
+    @objc private func refresh(_ refresh: UIRefreshControl) {
+        requestRefresh(.upcoming) {
+            refresh.endRefreshing()
+        }
+    }
+    
+    func reloadData() {
+        collectionView?.reloadData()
+    }
+}
+
+extension TMDBListViewController: UICollectionViewDelegate, UICollectionViewDelegateFlowLayout {
+    func collectionView(_ collectionView: UICollectionView,
+                        didSelectItemAt indexPath: IndexPath) {
+        defer { collectionView.deselectItem(at: indexPath, animated: false) }
+        delegate?.upcomingList(self, didSelect: datasource.items[indexPath.item])
+    }
+    
+    func scrollViewDidEndDragging(_ scrollView: UIScrollView,
+                                  willDecelerate decelerate: Bool) {
+        if scrollView.contentOffset.y < 0 { return }
+        if scrollView.contentOffset.y + scrollView.frame.height > scrollView.contentSize.height {
+            request.page += 1
+            makeRequest()
+        }
+    }
+    
+    func collectionView(_ collectionView: UICollectionView,
+                        layout collectionViewLayout: UICollectionViewLayout,
+                        sizeForItemAt indexPath: IndexPath) -> CGSize {
+        let portrait = UIDevice.current.orientation == .portrait
+        let padding: CGFloat = portrait ? 20 : 40
+        let collectionViewSize = collectionView.frame.size.width - padding
+        let columns: CGFloat = portrait ? 2.0 : 3.0
+        return CGSize(width: collectionViewSize/columns,
+                      height: collectionViewSize/columns)
+    }
+}
+
+extension TMDBListViewController: UISearchBarDelegate {
+    func searchBar(_ searchBar: UISearchBar,
+                   textDidChange searchText: String) {
+        let filteredItems = staticMovies.filter({ $0.title.contains(searchText) })
+        if filteredItems.isEmpty {
+            datasource.items = staticMovies
+        } else {
+            datasource.items = filteredItems
+        }
+        reloadData()
+    }
+    
+    func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        guard let searchText = searchBar.text else { return }
+        request = TMDBRequests(with: searchBar.text!.isEmpty ? .upcoming : .search(searchText))
+        makeRequest()
```

Comment:
While the characters are being typed, the class use the staticMovies variables to search for the objects that are already in the database instead of using network connection